examples/supply_chain/rl/env_sampler.py

```diff
@@ -143,75 +144,80 @@ def __init__(
-        extend_state(self._facility_in_transit_orders[entity.facility_id])
-        extend_state([self._product_id2idx[entity.facility_id][entity.skus.id] + 1])  # TODO: check +1 or not
-        extend_state([be.world.get_facility_by_id(entity.facility_id).get_max_vlt(entity.skus.id)])
-        extend_state([entity.skus.service_level])
-        return np.array(np_state + offsets)
-
-    def get_rl_policy_state(self, entity_id: int, entity: SupplyChainEntity) -> np.ndarray:
-        state = self._agent_states.update_entity_state(
+    def get_or_policy_state(self, entity: SupplyChainEntity) -> dict:
+        if self._storage_unit_cost_dict is None:
+            self._storage_unit_cost_dict, self._storage_capacity_dict = self._get_storage_unit_cost_and_capacity_info()
+
+        state = self._or_agent_states._update_entity_state(
+            entity_id=entity.id,
+            storage_unit_cost_dict=self._storage_unit_cost_dict,
+            storage_capacity_dict=self._storage_capacity_dict,
+            product_metrics=self._cur_metrics["products"].get(self._unit2product_unit[entity.id], None),
+            cur_consumer_hist_states=self._cur_consumer_hist_states,
+            product_levels=self._storage_product_quantity[entity.facility_id],
+            in_transit_order_quantity=self._facility_in_transit_orders[entity.facility_id],
+        )
+
+        return state
+
+    def get_rl_policy_state(self, entity_id: int) -> np.ndarray:
+        np_state = self._rl_agent_states.update_entity_state(
             entity_id=entity_id,
             tick=self._learn_env.tick,
             cur_metrics=self._cur_metrics,
             cur_distribution_states=self._cur_distribution_states,
-            cur_seller_states=self._cur_seller_states,
-            cur_consumer_states=self._cur_consumer_states,
+            cur_seller_hist_states=self._cur_seller_hist_states,
+            cur_consumer_hist_states=self._cur_consumer_hist_states,
             accumulated_balance=self._balance_calculator.accumulated_balance_sheet[entity_id],
             storage_product_quantity=self._storage_product_quantity,
             facility_product_utilization=self._facility_product_utilization,
             facility_in_transit_orders=self._facility_in_transit_orders,
         )
-        np_state = serialize_state(state)
         return np_state
 
-    def _get_entity_state(self, entity_id: int) -> np.ndarray:
+    def _get_entity_state(self, entity_id: int) -> Union[np.ndarray, dict]:
         entity = self._entity_dict[entity_id]
 
         if isinstance(self._policy_dict[self._agent2policy[entity_id]], RLPolicy):
-            return self.get_rl_policy_state(entity_id, entity)
-        else:
+            return self.get_rl_policy_state(entity_id)
+        elif isinstance(self._policy_dict[self._agent2policy[entity_id]], ConsumerBasePolicy):
             return self.get_or_policy_state(entity)
+        else:
+            return []
```

Comment:
Should not return a list.

Reply:
None?

Reply:
According to the return type, we should return a `np.ndarray` or a `dict`. If we want to return `None`, we could add `None` into the returned `Union`.